internal/declarative/v2/ready_check.go

```diff
@@ -24,16 +25,20 @@ func NewExistsReadyCheck() ReadyCheck {
 
 type ExistsReadyCheck struct{}
 
-func (c *ExistsReadyCheck) Run(ctx context.Context, clnt Client, _ Object, resources []*resource.Info) (State, error) {
+func (c *ExistsReadyCheck) Run(
```

Comment:
The `ExistsReadyCheck` type is actually only used in tests. Maybe it's worth to move it to some *_test package?